evalml/tests/component_tests/test_simple_imputer.py

```diff
@@ -626,3 +626,39 @@ def test_simple_imputer_boolean_nullable_valid_train_empty_test():
     X_t = imp.transform(X_test)
     assert not X_t["a"].isna().any()
     assert isinstance(X_t.ww.logical_types["a"], BooleanNullable)
+
+
+def test_simple_imputer_all_bools_at_fit_with_nans_at_transform():
+    # X_train will be only bool dtypes so the _component_obj won't be fit
+    X_train = pd.DataFrame(
+        {
+            "bools1": pd.Series([True, False, True, True] * 20),
+            "bools2": pd.Series([True, False, True, False] * 20),
+        },
+    )
+    X_train.ww.init(
+        logical_types={
+            "bools1": "Boolean",
+            "bools2": "Boolean",
+        },
+    )
+
+    imp = SimpleImputer()
```

Comment:
A lot of our other tests parameterize the imputation strategy - it'd be great (and I think would have caught what Jeremy mentioned above) to do the same here and test all the boolean imputation strategies!

Reply:
Given how `mean` and `median` can raise a TypeConversionError for boolean data, I'm going to treat them as incompatible with boolean data for now, so I'm not going to test with them. I will, however, make the use of `most_frequent` explicit.